EIPS/eip-####.md

```diff
@@ -0,0 +1,124 @@
+
+## Motivation
+
+The Git version control system can be used as a blockchain. The individual commits represent execution blocks and the act of pointing the `mainnet` branch to a specific commit represents consensus fork choice.
+
+Switching the Ethereum engine to Git leverages pre-existing functionality, avoiding the need to reinvent wheels. Such a simplification not only increases security by reducing the attack scope, but also avoids frustration from failed testnet upgrades, complex syncing, multi-year data sharding plans and so on.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Storage
+
+The execution state is tracked in the `mainnet` branch of a Git repository. Each account is represented by its own directory containing a `BALANCE` file with its balance, a `NONCE` file with its account nonce, and a `CODE` file with its EVM code (if present). Storage slots are represented with additional files in subdirectories.
+
+### Accounts
+
+Account addresses are deterministically derived from the GitHub account ID. The directory name is derived by separating the address in chunks of 4 bytes (8 letters each). For example, the account `0x0001020304050607080910111213141516171819` would be stored in directory `00010203/04050607/08091011/12131415/16171819`.
+
+### Transactions
+
+Transactions are created using a GitHub issue template. On creation of an issue, a GitHub action is triggered that eventually executes the transaction and updates the state by creating a commit to the Git repository.
+
+The commit message backlinks to the issue, and the issue is closed as "completed" with a comment containing the corresponding receipt data. For each log, the GitHub action further adds a label to the issue for the log-emitting address and indexed topics.
+
+The GitHub issue template follows the usual transaction fields (including `To`, `Value`, `Input`), with the following differences.
+
+#### Signature
+
+No signature is needed. The `From` address is derived from the GitHub account opening the issue. The `Nonce` is obtained from the current state when the transaction executes. Transactions are executed in ascending issue number order.
+
+If a transaction fails as it lacks the required `BALANCE`, the issue is closed as "not completed".
+
+#### Fee market
+
+The base fee is set based on a moving average tracking the transaction execution time as given by the time difference of the GitHub issue being opened and it being closed by the system.
+
+There are no priority fees. Transactions are always executed in order of submission.
+
+Spam transactions are implicitly solved at the discretion of GitHub. Users creating lots of nonsensical GitHub issues MAY get banned and can contact GitHub support for followup.
```

Comment:
This ensures we are censorship resistant legally 👍

Reply:
https://ethereum-magicians.org/t/eip-gitchain/23336/8?u=etan-status